search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx

```diff
@@ -92,10 +92,19 @@ export default class SearchVerticalsContainer extends React.Component<ISearchVer
       const vertical = this.props.verticals[verticalIdx];
       if (vertical.isLink) {
           // Send the query to the new page
-          const behavior = vertical.openBehavior === PageOpenBehavior.NewTab ? '_blank' : '_self';
           this.props.tokenService.resolveTokens(vertical.linkUrl).then((resolvedUrl: string) => {           
             resolvedUrl = resolvedUrl.replace(/\{searchTerms\}|\{SearchBoxQuery\}/gi, GlobalSettings.getValue(BuiltinTokenNames.inputQueryText));
-            window.open(resolvedUrl, behavior);
+
+            if(vertical.openBehavior === PageOpenBehavior.NewTab){
+              window.open(resolvedUrl, "_blank");
+            }else{
+              // Allow SharePoint to intercept the click and do a soft navigation
+              document.body.insertAdjacentHTML('beforeend', `<a href="${resolvedUrl}" style="display:none;"></a>`);
+              const anchor = document.body.lastElementChild as HTMLElement; 
+              anchor.click();
+              document.body.removeChild(anchor);
```

Comment:
Any chance of getting an error here if the navigation happens before the DOM is removed? Adding a try/catch around removeChild ensures that this case never happens.

Reply:
This is unlikely to happen, as the code block should run synchronously before the click event is resolved.
But in any case, we can validate if the anchor.parentElement is not null before removing, which would be a valid operation even if it weren't in the DOM somehow, and wouldn't result in an error.

```suggestion
              anchor.parentElement?.removeChild(anchor);
```